lib/linter.js

```diff
@@ -74,12 +74,19 @@ module.exports = function (cwd, args, text) {
 
     return JSON.stringify(fileConfig, null, '  ');
   }
+  if (currentOptions.fixToStdout && !stdin) {
+    return 'The --fix-to-stdout option must be used with --stdin.'
+      + '\n# exit 1';
+  }
   var report;
   if (stdin && text) {
     report = engine.executeOnText(text, currentOptions.stdinFilename);
   } else {
     report = engine.executeOnFiles(files);
   }
+  if (currentOptions.fixToStdout) {
+    return report.results[0].output || text;
```

Comment:
Maybe the new option should only allow a single file if not used with `--stdin`? This way we don't need to require the `--stdin` option and one could use the feature to "inspect" what eslint does with `--fix` without actually modifying the file.